src/utils.js

```diff
@@ -132,7 +132,10 @@ export function sanityCheck(
   if (hash) {
     const auth = verifyFn(password, users[user]);
     if (auth) {
-      return true;
```

Comment:
If you remove this, the user will never be logged.

Reply:
npm expects the 409 error and handles it, continuing the login process calling another endpoint. The user will login. We could return the same data as when the user is first created, but this behaviour would differ from the standard registry behaviour. couchdb will return a 409 in this case because a revision is not sent with the request ([link](https://wiki.apache.org/couchdb/HTTP_Document_API)). Returning true will cause npm to fail as we will return a 200 with body "true". While this is a valid JSON, npm will attempt to set the username on the response, which will cause an error (as the response is a boolean) and the user will not login. Below is the relevant code from [npm](https://github.com/npm/npm-profile/blob/latest/lib/index.js):
```
function loginCouch (username, password, conf) {
  validate('SSO', arguments)
  const userobj = {
    _id: 'org.couchdb.user:' + username,
    name: username,
    password: password,
    type: 'user',
    roles: [],
    date: new Date().toISOString()
  }
  const logObj = {}
  Object.keys(userobj).forEach(k => {
    logObj[k] = k === 'password' ? 'XXXXX' : userobj[k]
  })
  process.emit('log', 'verbose', 'login', 'before first PUT', logObj)

  const target = url.resolve(conf.registry, '-/user/org.couchdb.user:' + encodeURIComponent(username))
  return fetchJSON(Object.assign({method: 'PUT', target: target, body: userobj}, conf)).catch(err => {
    if (err.code === 'E400') {
      err.message = `There is no user with the username "${username}".`
      throw err
    }
    if (err.code !== 'E409') throw err
    return fetchJSON(Object.assign({method: 'GET', target: target + '?write=true'}, conf)).then(result => {
      Object.keys(result).forEach(function (k) {
        if (!userobj[k] || k === 'roles') {
          userobj[k] = result[k]
        }
      })
      const req = {
        method: 'PUT',
        target: target + '/-rev/' + userobj._rev,
        body: userobj,
        auth: {
          basic: {
            username: username,
            password: password
          }
        }
      }
      return fetchJSON(Object.assign({}, conf, req))
    })
  }).then(result => {
    result.username = username
    return result
  })
}
```

Reply:
You are right @colombotapps 👍 good research
![screen shot 2018-03-30 at 9 38 49 pm](https://user-images.githubusercontent.com/558752/38151254-cb4dea9e-3462-11e8-8fea-7c014b42af57.png)

```
Email: (this IS public) (dsa@dasd.com) dsadsa@das.com
npm ERR! Cannot create property 'username' on boolean 'true'

npm ERR! A complete log of this run can be found in:
```